File: backend/src/models/task.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional, TYPE_CHECKING
from src.db.base import TimestampMixin

if TYPE_CHECKING:
    from src.models.user import User


class Task(TimestampMixin, table=True):
    """Task model for todo items"""
    __tablename__ = "tasks"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="users.id", index=True)
    title: str = Field(max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    status: str = Field(default="Incomplete", max_length=50)

    # Relationship
    user: "User" = Relationship(back_populates="tasks")

File: shared/types/task.ts
/**
 * Task Type Definition
 * Matches backend Task model from src/models/task.py
 */

export type TaskStatus = "Complete" | "Incomplete";

export interface Task {
  id: number;
  user_id: number;
  title: string; // 1-200 characters
  description: string | null; // 0-1000 characters, optional
  status: TaskStatus;
  created_at: string; // ISO 8601 datetime string
  updated_at: string; // ISO 8601 datetime string
}

export interface TaskCreate {
  title: string;
  description?: string;
}

export interface TaskUpdate {
  title?: string;
  description?: string;
}

export interface TaskListResponse {
  tasks: Task[];
  total: number;
}
